reproducible/docker/Dockerfile.buster

```diff
@@ -0,0 +1,26 @@
+FROM debian:buster
+
+RUN apt-get update && apt-get upgrade -y -o Dpkg::Options::="--force-confold" --no-install-recommends
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends \
+  faketime build-essential curl gccgo git
+
+# libgtk2.0-dev libgtk-3-dev gtk2.0 gtk+3.0
+RUN DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends \
+  libgtk-3-dev ruby
+
+RUN update-alternatives --install /usr/bin/go go /usr/bin/go-7 1
+
+# Clean up APT when done.
+RUN apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+ADD setup-reproducible.sh /root/setup-reproducible
+ADD download-golang-1.7.sh /root/download-golang
```

Comment:
Since you are there, would you like to update Golang to version 1.9? It is the current stable and could have some 

But I wonder which version of golang is used by the Debian project to test if a package is reproducible.

Reply:
> But I wonder which version of golang is used by the Debian project to test if a package is reproducible.

It depends on the target debian version the package is build for:

```
golang     | 2:1.7~5           | stable             | amd64, arm64, armel, armhf, i386, ppc64el
golang     | 2:1.8~3           | testing            | amd64, arm64, armel, armhf, i386, ppc64el
golang     | 2:1.9~1           | unstable           | amd64, arm64, armel, armhf, i386, ppc64el
```

The best would be not to install any additional go version but to use the debian  packaged one, so they don't differ from https://tests.reproducible-builds.org/debian/rb-pkg/buster/amd64/coyim.html.

Reply:
Related to this, I believe our goal for reproducibility is the less strict form (from https://www.debian.org/doc/debian-policy/#reproducibility):

> repeatedly building the source package for the build architecture on any machine of the host architecture with those versions of the build dependencies installed and exactly those environment variable values set will produce bit-for-bit identical binary packages.

and not the more strict form

> [...] that packages produce bit-for-bit identical binaries even if most environment variables and build paths are varied.

Do you think is worth making this distinction clear in `REPRODUCIBLE.md`? Is it clear for you from reading the document?

Reply:
I'd like to overhaul the `REPRODUCIBLE.md` when we're finish with this goal (which is still not entirely clear to me).
In order to fix the unreproducibility (what a word) for the buster debian package, we would need to be compliant with the strict form, because that's what debian is testing when building the package.

Reply:
We decided to not rely on the golang provided by the distro but build it reproducibly ourselves - again, by the less strict definition of reproducible.

If we always build golang from the same source path, it's reproducible by the less strict definition and golang/go#16860 is less relevant (source/build paths are deterministic - maybe not build path).

But I also wonder what is the recommended and pragmatic approach from Debian project to release reproducibly now.

Reply:
The buster image is mainly to debug why the coyim debian package build is currently unreproducible (see https://tests.reproducible-builds.org/debian/rb-pkg/buster/amd64/coyim.html). 
This is why I think it's good to have the same [Installed-Build-Depends as used by debian](https://tests.reproducible-builds.org/debian/buildinfo/buster/amd64/coyim_0.3.8+ds-4_amd64.buildinfo) (which is the packaged golang-1.8 deb).

Can you explain me what's the difference between the packaged version and the built from source one is again, because I don't understand your comment above ? And how is that related to golang/go#16860 ?

Reply:
Oh, maybe this is because go 1.8 on debian might produce a differnt binary than go 1.8 on distribuition X/Y ?

Reply:
golang/go#16860 may be the wrong issue. Sorry.

By the time we started this, there was a lot of issues on different problems with having reproducible builds on Golang (even in the less strict sense). This may give you some historical idea: https://blog.filippo.io/reproducing-go-binaries-byte-by-byte/

The general idea was by the time was: build `go` from a controlled source path and to a controlled `GOROOT`. And then, build your app reproducibly using this controlled `go` environment.